### PR TITLE
feat(rpc): improve error message for nonce too low

### LINKED ISSUE #10498: Improve rpc error for nonce too low
### Describe the feature

ref https://github.com/paradigmxyz/reth/issues/10464

the current error isn't very helpful: `nonce too low`

ideally we want:

> nonce too low: next nonce 72, tx nonce 71

the revm type even has this already:

```
    NonceTooLow {
        tx: u64,
        state: u64,
    },
```

so it's just a matter of integrating this in the rpc error:

https://github.com/paradigmxyz/reth/blob/8df9efd6ca745886a349d350011a354de56b0bef/crates/rpc/rpc-eth-types/src/error.rs#L265-L267

### Additional context

cc @ShantelPeters

_No response_

### PR DESCRIPTION
**Pull request by i-sangh**
ref #10464. closes https://github.com/paradigmxyz/reth/issues/10498.

The current error wasn't very helpful: nonce too low

So ideally it should now be:

nonce too low: next nonce 72, tx nonce 71

### CODE REVIEW

File: crates/rpc/rpc-eth-types/src/error.rs [hunk 1]
```diff
@@ -264,7 +268,12 @@ where
 pub enum RpcInvalidTransactionError {
     /// returned if the nonce of a transaction is lower than the one present in the local chain.
     #[error("nonce too low")]
```
Comment (emhane) [inline crates/rpc/rpc-eth-types/src/error.rs:270]:
```suggestion
    #[error("nonce too low: got {tx}, expected {state}")]
```

**Review by emhane — CHANGES_REQUESTED** [review]

**Comment by emhane** [comment]
make sure to run lint with `nightly` tool chain, otherwise a bunch of semicolons are added that we don't want, try command `make lint`

**Review by emhane — APPROVED** [review]

File: crates/rpc/rpc-eth-types/src/error.rs [hunk 2]
```diff
@@ -494,7 +503,7 @@ impl From<reth_primitives::InvalidTransactionError> for RpcInvalidTransactionErr
         // txpool (e.g. `eth_sendRawTransaction`) to their corresponding RPC
         match err {
             InvalidTransactionError::InsufficientFunds { .. } => Self::InsufficientFunds,
-            InvalidTransactionError::NonceNotConsistent => Self::NonceTooLow,
+            InvalidTransactionError::NonceNotConsistent => Self::NonceTooLow { tx: 0, state: 0},
```
Comment (mattsse) [inline crates/rpc/rpc-eth-types/src/error.rs:506]: I think we can also add these fields to the `NonceNotConsistent` error variant

https://github.com/paradigmxyz/reth/blob/29058ad7ee64d993de6c301a9d688002cb698f03/crates/transaction-pool/src/validate/eth.rs#L338-L344

File: crates/rpc/rpc-eth-types/src/error.rs [hunk 3]
```diff
@@ -236,7 +236,11 @@ where
 {
     fn from(err: EVMError<T>) -> Self {
         match err {
-            EVMError::Transaction(err) => RpcInvalidTransactionError::from(err).into(),
+            
+            EVMError::Transaction(_) => EthApiError::InvalidTransaction(
+                RpcInvalidTransactionError::NonceTooLow { tx: 0, state: 0 }
+            ),
```
Comment (mattsse) [inline crates/rpc/rpc-eth-types/src/error.rs:242]: this doesn't look right, why this change?

**Review by mattsse — CHANGES_REQUESTED** [review]
one more suggestion

**Comment by i-sangh** [comment]
error.rs & eth.rs

**Comment by mattsse** [comment]
why was this closed @i-sangh ? this was on track and almost ready

**Comment by emnul** [comment]
Happy to fork and continue work on on this one @mattsse

**Pull request closed without merging**